infra/cifuzz/clusterfuzz_deployment.py

```diff
@@ -99,17 +100,29 @@ def download_latest_build(self):
       # called if multiple bugs are found.
       return self.workspace.clusterfuzz_build
 
-    _make_empty_dir_if_nonexistent(self.workspace.clusterfuzz_build)
-    build_name = self._get_build_name()
+    repo_dir = self.ci_system.repo_dir()
+    if not repo_dir:
+      raise RuntimeError('repo checkout does not exist.')
 
-    try:
-      logging.info('Downloading latest build.')
-      if self.filestore.download_build(build_name,
-                                       self.workspace.clusterfuzz_build):
-        logging.info('Done downloading latest build.')
-        return self.workspace.clusterfuzz_build
-    except Exception as err:  # pylint: disable=broad-except
-      logging.error('Could not download latest build because of: %s', err)
+    _make_empty_dir_if_nonexistent(self.workspace.clusterfuzz_build)
+    repo = repo_manager.RepoManager(repo_dir)
+
+    # Builds are stored by commit, so try the latest |LATEST_BUILD_WINDOW|
+    # commits before the current.
+    for old_commit in repo.get_commit_list('HEAD^',
```

Comment:
Maybe download build in filestore needs to take a list of names to download and then download the first one. Then we can pass it a large list of commits and not worry about doing a lot of API calls (as this approach does).

Reply:
I can add a TODO here. Is API usage limited/a concern here? This is only 3x in the worst case (and in most cases should just be a single API call because a previous commit build generally should complete before a fuzzing job for the current commit.).